### ISSUE TITLE
Personal Nameplate Health Bar is incorrectly hidden when you die

### ISSUE THREAD

**Issue by millanzarreta** [Stanzilla/WoWUIBugs#397]
This bug begins with the Dragonflight exp, and still happening in the current 10.0.5 version. The personal nameplate health bar is incorrectly hidden when you die, something like this:

![wow_personalnameplate_healthbar_bug](https://user-images.githubusercontent.com/8227951/222867944-57b69ca8-f201-48e1-a3fd-7e46051608a8.gif)

The main issue is that the ``CompactUnitFrame_UpdateHealthColor`` function in the ``CompactUnitFrame.lua`` file checks if the unit is dead, and if it is dead it sets the flag variable ``hideHealthbar`` to ``true``, and it does this for all nameplates, included for the personal nameplate, which doesn't seem to be the correct behaviour:

```lua
...
function CompactUnitFrame_UpdateHealthColor(frame)
	local r, g, b;
	if ( not UnitIsConnected(frame.unit) ) then
		--Color it gray
		r, g, b = 0.5, 0.5, 0.5;
	elseif (UnitIsDead(frame.unit)) then
		--Color it gray
		r, g, b = 0.5, 0.5, 0.5;
		-- Also hide the health bar
		frame.hideHealthbar = true;
	else
		...
	end
	...
end
...
```

Subsequently, in other functions, the visibility of the health bar of the nameplate is established according to what is indicated by this ``hideHealthbar`` flag. For example, in my tests, the most common thing is that the ``CompactUnitFrame_UpdateWidgetsOnlyMode`` function hides the health bar when the following line is executed:

``frame.healthBar:SetShown(not inWidgetsOnlyMode and not frame.hideHealthbar);``

In Dragonflight the ``CompactUnitFrame_UpdateHealthColor`` function was modified to add this section of code:

![imagen](https://user-images.githubusercontent.com/8227951/222863870-8bec5ed5-8870-40b2-ae98-79db966142b5.png)

I think this was done by forgetting about the personal nameplate, for which you should never hide the health bar, even if the player is dead.

This bug doesn't always happen, and it tends to happen mostly when I'm doing dungeons, and not so much when you're alone in the world. I think this depends on the order and frequency in which the events are executed.

Finally, to reproduce the bug, I think the best is to enter into a dungeon, surround yourself with enemies and die. It may not always happen, but eventually, it should. I'm using a max-level character in DF dungeons, and, in case it helps, my _CVars_ related to personal nameplates are currently the following, although they don't seem to be related to this bug:

```
NameplatePersonalClickThrough		1
NameplatePersonalHideDelayAlpha		0.45
NameplatePersonalHideDelaySeconds	3
NameplatePersonalShowAlways		0
NameplatePersonalShowInCombat		1
NameplatePersonalShowWithTarget		0
nameplateShowPersonalCooldowns		0
```

**Comment by millanzarreta** [comment]
In case it's helpful to anyone, the temporal workaround that I use to avoid this bug is the following, a simple hook to the ``CompactUnitFrame_UpdateWidgetsOnlyMode`` function, which after checking it I think is the only function that takes into account the ``hideHealthbar`` flag to hide the health bar:

```lua
local strfind = string.find
local playerGUID = UnitGUID("player")
hooksecurefunc("CompactUnitFrame_UpdateWidgetsOnlyMode", function(frame)
	if ((frame.WidgetContainer ~= nil) and frame.hideHealthbar and strfind(frame.unit, "nameplate") and (UnitGUID(frame.unit) == playerGUID)) then
		if not(frame.healthBar:IsShown()) then frame.healthBar:Show() end
	end
end)
```
Maybe not the best workaround, but at least it seems to work.

**Comment by Meorawr** [comment]
Fixed in 10.1.0.48537.
